добавлена обработка параметра КаталогиБиблиотек при старте ОФ из командной строки

Golovanoff opened the pull request:
реализация  #

сделанные изменения
при старте ОФ из командной строки:
добавлена обработка параметра КаталогиБиблиотек
исправлена проверка параметров json - сравнения со строкой "Истина" исправлены на булево  Истина
добавлена возможность указания относительных путей (от "." или $workspaceRoot) в строках массивов json (в частности, КаталогиБиблиотек)
-
-

@vanessa-opensource/Collaborators - просьба прокомментировать и проверить

Commit:
добавлена обработка параметра КаталогиБиблиотек при запуске ОФ из командной строки

Commit:
изменения в бинарнике формы

artbear commented on epf/bddRunner/bddRunner/Forms/Форма/Ext/Form/Module.bsl:
```diff
@@ -2270,14 +2292,14 @@
+		ФайлПути = Новый Файл(ОригСтр);
 		//Для совместимости со старым кодом оставим пока КаталогИнструментов, правильней было-бы использовать КаталогПроекта
 		ФайлПути = Новый Файл(ДополнитьСлешВПуть(КаталогИнструментов) + ОригСтр);
+		Возврат ФайлПути.ПолноеИмя;
```
сразу бросилось в глаза, что одна из установок лишняя - `ФайлПути = XXX`
```
ФайлПути = Новый Файл(ОригСтр); // 1
//Для совместимости со старым кодом оставим пока КаталогИнструментов, правильней было-бы использовать КаталогПроекта
ФайлПути = Новый Файл(ДополнитьСлешВПуть(КаталогИнструментов) + ОригСтр); // 2
Возврат ФайлПути.ПолноеИмя;
```

artbear commented on epf/bddRunner/bddRunner/Forms/Форма/Ext/Form/Module.bsl:
```diff
@@ -2258,8 +2278,10 @@
 Функция ПреобразоватьПутьСТочкамиКНормальномуПути(Знач ОригСтр="") Экспорт
 
-	Если ПустаяСтрока(ОригСтр) Тогда
-		Возврат ОригСтр;
```
проверку на пустоту желательно бы оставить.

artbear commented on epf/bddRunner/bddRunner/Forms/Форма/Ext/Form/Module.bsl:
```diff
@@ -2258,8 +2278,10 @@
+		Для Сч = 0 По ОригСтр.Количество() - 1 Цикл
+			ОригСтр[Сч] = ПреобразоватьПутьСТочкамиКНормальномуПути(ОригСтр[Сч]);
+		КонецЦикла;
 	КонецЕсли;
```
похоже, что здесь зря нет Возврата для параметра-массив

ведь дальнейшая обработка в коде идет к строке, а на массив не рассчитана (

например `Если Найти(ОригСтр, "$workspaceRoot") > 0 Тогда` выдаст ошибку для массива

artbear commented on epf/bddRunner/bddRunner/Forms/Форма/Ext/Form/Module.bsl:
```diff
@@ -398,13 +398,28 @@
+	Если ТипЗнч(ЗначениеИзКонфига) = Тип("Массив") Тогда
+		Нужно = Ложь;
+		Для Каждого ЗначениеСписка Из ЗначениеИзКонфига Цикл
+			Нужно = Нужно ИЛИ НужноПреобразоватьПуть(ЗначениеСписка);
```
Если очередной вызов `НужноПреобразоватьПуть(ЗначениеСписка)` это Истина, 

то лучше сразу сделать `Возврат Истина`, а не обрабатывать цикл дальше.

artbear commented on epf/bddRunner/bddRunner/Forms/Форма/Ext/Form/Module.bsl:
```diff
@@ -479,20 +494,25 @@
 
 		ЗнДелатьСообщенияТранслитом = ПолучитьЗначениеПереданногоПараметра(СтрутктураJSON,"ДелатьСообщенияТранслитом");
-		Если ЗнДелатьСообщенияТранслитом = "Истина" Тогда
+		Если ЗнДелатьСообщенияТранслитом = Истина Тогда
```
а точно метод ПолучитьЗначениеПереданногоПараметра умеет преобразовывать строковые значения "Истина"\"Ложь" в Истина\Ложь или нет?

ведь если не умеет, тогда ты вносишь несовместимость со старым форматом настроек (

и нужно решать, что с делать с этой поддержкой - оставлять или убирать?

artbear reviewed (changes requested):
есть замечания.

Golovanoff committed:
оптимизация НужноПреобразоватьПуть 

теперь, если если найден первый элемент массива,  который необходимо преобразовывать, функция сразу возвращает истину, не обрабатывая оставшиеся элементы

Golovanoff committed:
ПреобразоватьПутьСТочкамиКНормальномуПути возвращает массив сразу

теперь функция ПреобразоватьПутьСТочкамиКНормальномуПути сразу возвращает массив после обработки

Golovanoff committed:
Merge branch 'vanessa-opensource:develop' into develop
